### PR TITLE
Build on top of salt doc updates

### LINKED ISSUE #3847: Document how to build a custom component using Salt theme
### Area

Documentation

### The problem

User often needs to create custom components to play nicely with Salt components, currently it's very hard to navigate the "how"

### The solution

Document how to use existing tokens to create custom components, using some representative tokens (e.g. background, foreground, spacing, size). Improve theming / characteristics documentation ([token page](https://www.saltdesignsystem.com/salt/theming/accordionview)) may also be needed.

### Alternatives and examples

.

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by mark-tate**
Closes #3847

### CODE REVIEW

File: site/docs/theming/best-practices.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+Salt provides pre-made, system-wide [styling options](/salt/theming/index#styling-options) that users can opt into through the SaltProvider. Salt will continue to expand this offering, but it currently offers the easiest, cheapest, and most stable way to customize your UI.
+
+### Option 3: Extend the Salt theme
+
+Each Salt theme consists of 3 layers ([Foundation](/salt/about/glossary#foundation), [Palette](/salt/about/glossary#palette-layer) and [Characteristics](/salt/about/glossary#characteristics-layer) layers) that work together to offer a consistent theming solution across all components:
+
+Whilst this creates the highest level of flexibility it comes with an increased level of responsibility to maintain. If this is what you need , then here are our recommendations:
+
+#### Don'ts
```
Comment (origami-z) [inline site/docs/theming/best-practices.mdx:44]: Dos and Don't tells a few different opinions from Salt, needs very careful team-wide review

File: site/docs/theming/best-practices.mdx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+#### Dos
+
+- Do create your own theme that references the Salt characteristics and palette layers to provide a holistic theming solution.
+- Do document your design language so others understand the design intent and how to use it. This avoids future duplication and eases updates.
+- Do change the palette layer but federate the solution to a central theme, ensuring that deviations from the Salt standard are known and can be easily updated if required.
+- Do change the characteristics layer but federate the solution to a central theme, ensuring that deviations from the Salt standard are known and can be easily updated if required.
+- Do review the Salt backlog/roadmap before making a local change, to understand whether your change is tactical or strategic
+
+We've documented all our [design tokens](/salt/theming/index#design-tokens), but let's start with some basic tokens that have the biggest effect:
```
Comment (origami-z) [inline site/docs/theming/best-practices.mdx:57]: > let's start with some basic tokens that have the biggest effect

This feels the encouragement of changing these values?

File: site/docs/theming/best-practices.mdx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+| `--salt-content-primary-foreground`         | **Salt's default foreground color**<br/><br/>This token serves as the [default foreground color](https://github.com/jpmorganchase/salt-ds/blob/e76e5b92dcb6cfae02813137098393b91f54e501/packages/theme/css/global.css#L6) for text and icons, ensuring consistency across your UI.        |
+| `--salt-status-info-foreground-informative` | **Salt info status foreground**<br/><br/>The `informative` suffix ensures this token meets the minimal contrast ratio when used against Salt container background colors, making it suitable for informative text (as opposed to `decorative`).                                           |
+| `--salt-accent-borderColor`                 | **Salt's accent styling option**<br/><br/>Use this token for border colors to reflect the accent styling option choice – `teal` or `blue`. The `borderColor` naming convention helps maintain clarity and consistency.                                                                    |
+
+<br />
+These four variables represent key aspects of theming controls: background,
+foreground, typography, and border color. Exploring these variables will
```
Comment (origami-z) [inline site/docs/theming/best-practices.mdx:68]: > Exploring these variables

We shouldn't encourage change these

**Comment by origami-z** [comment]
This adds too many new recommendations to the original intend of basic educational purpose, which means eating into product roadmap and #4004. I think we should close this entirely and rework piece by piece

**Comment by origami-z** [comment]
Back to draft while working out #3847

**Comment by mark-tate** [comment]
will update and revisit in line with 2025 direction at a later date

**Pull request closed without merging**